Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1038,7 +1058,43 @@ - (void)_didCommitLayerTree:(const WebKit::RemoteLayerTreeTransaction&)layerTree
             _page->findClient().didAddLayerForFindOverlay(_page.get(), layer);
         }
     }
+
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+    [self _updateOverlayRegions:layerTreeTransaction.changedLayerProperties() destroyedLayers:layerTreeTransaction.destroyedLayers()];
+#endif
+}
+
+#if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
+- (void)_updateOverlayRegions:(const WebKit::RemoteLayerTreeTransaction::LayerPropertiesMap&)changedLayerProperties destroyedLayers:(const Vector<WebCore::GraphicsLayer::PlatformLayerID>&)destroyedLayers
+{
+    if (!changedLayerProperties.size() && !destroyedLayers.size())
+        return;
+
+    const auto& layerTreeHost = downcast<WebKit::RemoteLayerTreeDrawingAreaProxy>(*_page->drawingArea()).remoteLayerTreeHost();
+    auto overlayRegionsWithIDs = [_scrollView overlayRegionsWithIDs];
+    auto existingOverlayRegionsWithIDsSize = overlayRegionsWithIDs.size();
+
+    overlayRegionsWithIDs.removeIf([&layerTreeHost](const auto& it) {
+        return !layerTreeHost.layerForID(it.key);
```

Comment:
This removal should be done after `fixedIDs` is pruned otherwise `overlayRegionsWithIDs` can contains layers which are no longer fixed.